ner_v2/detectors/numeral/number/zh-TW/number_detection.py

```diff
@@ -149,4 +151,73 @@ def get_number_digit_by_digit(self, text=''):
 
     def get_number_with_digit_scaling(self, text=''):
         # change the below logic to work with scaling
-        return text
\ No newline at end of file
+        result = ''
+        if not text:
+            return result
+
+        # extract the digits and scales from the text
+        digit_list = []
+
+        pwr_index_map = {}
+        for t in text:
+            digit_val = self.base_numbers_map_full.get(t)
+            if digit_val is not None:
+                digit_list.append(digit_val)
+                if digit_val in self.power_of_10:
+                    pwr_index_map[digit_val] = len(digit_list) - 1
+            else:
+                return result
+        if len(digit_list) == 0:
+            return result
+
+        pwr_index_map[1] = len(digit_list)
+
+        # starting from highest power aggregate digit and scales
+        pwr_index_list = sorted(pwr_index_map.items(), key=lambda kv: kv[0], reverse=True)
+
+        st = 0
+        final_val = 0
+        for pwr, indx in pwr_index_list:
+            value = self.combine_digit_and_scale(digit_list[st: indx])
+            st = indx + 1
+            if value is not None:
+                value = value * pwr
+                final_val += value
+
+        result = str(final_val)
+        return result
+
+    def combine_digit_and_scale(self, num_list: None):
```

Comment:
we are assigning the type to the default value.

Reply:
this is also fixed